### PR TITLE
Add Ansible and Bash remediations for directory_access_var_log_audit rule

### PR DESCRIPTION
**Pull request by redhatrises**

### CODE REVIEW

File: linux_os/guide/system/auditing/auditd_configure_rules/directory_access_var_log_audit/ansible/shared.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  lineinfile:
+    path: '{{ all_files[0] }}'
+    line: -a always,exit -F dir=/var/log/audit/ -F perm=r -F auid&gt;=1000 -F auid!=unset
+      -F key=access-audit-trail
+    create: true
+
+- name: Inserts/replaces the /var/log/audit/ rule in audit.rules
+  lineinfile:
+    path: /etc/audit/audit.rules
+    line: -a always,exit -F dir=/var/log/audit/ -F perm=r -F auid&gt;=1000 -F auid!=unset
```
Comment (shawndwells) [inline linux_os/guide/system/auditing/auditd_configure_rules/directory_access_var_log_audit/ansible/shared.yml]: The Ansible templates use ``-F auid >=1000``, e.g. 

`````
grep -rin "auid"  template_ANSIBLE_*                                     131  19:15:49 
template_ANSIBLE_audit_rules_dac_modification:40:    line: "-a always,exit -F arch=b32 -S {{{ ATTR }}} -F auid>={{{ auid }}} -F auid!=unset -F key=perm_mod"
template_ANSIBLE_audit_rules_dac_modification:46:    line: "-a always,exit -F arch=b64 -S {{{ ATTR }}} -F auid>={{{ auid }}} -F auid!=unset -F key=perm_mod"
template_ANSIBLE_audit_rules_dac_modification:54:    line: "-a always,exit -F arch=b32 -S {{{ ATTR }}} -F auid>={{{ auid }}} -F auid!=unset -F key=perm_mod"
template_ANSIBLE_audit_rules_dac_modification:60:    line: "-a always,exit -F arch=b64 -S {{{ ATTR }}} -F auid>={{{ auid }}} -F auid!=unset -F key=perm_mod"
template_ANSIBLE_audit_rules_file_deletion_events:40:    line: "-a always,exit -F arch=b32 -S {{{ NAME }}} -F auid>={{{ auid }}} -F auid!=unset -F key=delete"
template_ANSIBLE_audit_rules_file_deletion_events:46:    line: "-a always,exit -F arch=b64 -S {{{ NAME }}} -F auid>={{{ auid }}} -F auid!=unset -F key=delete"
template_ANSIBLE_audit_rules_file_deletion_events:54:    line: "-a always,exit -F arch=b32 -S {{{ NAME }}} -F auid>={{{ auid }}} -F auid!=unset -F key=delete"
template_ANSIBLE_audit_rules_file_deletion_events:60:    line: "-a always,exit -F arch=b64 -S {{{ NAME }}} -F auid>={{{ auid }}} -F auid!=unset -F key=delete"
template_ANSIBLE_audit_rules_path_syscall:43:    - "-a always,exit -F arch=b32 -S {{{ SYSCALL }}} -F a2&03 -F path={{{ PATH }}} -F auid>={{{ auid }}} -F auid!=unset -F key=modify"
template_ANSIBLE_audit_rules_path_syscall:51:    - "-a always,exit -F arch=b64 -S {{{ SYSCALL }}} -F a2&03 -F path={{{ PATH }}} -F auid>={{{ auid }}} -F auid!=unset -F key=modify"
template_ANSIBLE_audit_rules_path_syscall:62:    - "-a always,exit -F arch=b32 -S {{{ SYSCALL }}} -F a2&03 -F path={{{ PATH }}} -F auid>={{{ auid }}} -F auid!=unset -F key=modify"
template_ANSIBLE_audit_rules_path_syscall:71:    - "-a always,exit -F arch=b64 -S {{{ SYSCALL }}} -F a2&03 -F path={{{ PATH }}} -F auid>={{{ auid }}} -F auid!=unset -F key=modify"
`````

Which should be used?

Comment (redhatrises) [reply linux_os/guide/system/auditing/auditd_configure_rules/directory_access_var_log_audit/ansible/shared.yml]: @shawndwells fixed

File: linux_os/guide/system/auditing/auditd_configure_rules/directory_access_var_log_audit/ansible/shared.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  set_fact:
+    all_files:
+    - /etc/audit/rules.d/access-audit-trail.rules
+  when: find_var_log_audit.matched == 0
+
+- name: Use matched file as the recipient for the rule
+  set_fact:
+    all_files:
+    - '{{ find_var_log_audit.files | map(attribute=''path'') | list | first }}'
+  when: find_var_log_audit.matched &gt; 0
```
Comment (yuumasato) [inline linux_os/guide/system/auditing/auditd_configure_rules/directory_access_var_log_audit/ansible/shared.yml]: There is no need to escape `>` here.
```
TASK [Use matched file as the recipient for the rule] *************************************************************************************************************************************************************
fatal: [192.168.122.84]: FAILED! => {"msg": "The conditional check 'find_var_log_audit.matched &gt; 0' failed. The error was: template error while templating string: unexpected char '&' at 33. String: {% if find_var_log_audit.matched &gt; 0 %} True {% else %} False {% endif %}\n\nThe error appears to have been in '/home/wsato/git/scap-security-guide/tests/logs/rule-custom-2019-03-06-1019/xccdf_org.ssgproject.content_rule_directory_access_var_log_audit.yml': line 51, column 7, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n      - low_disruption\n    - name: Use matched file as the recipient for the rule\n      ^ here\n"}
	to retry, use: --limit @/home/wsato/git/scap-security-guide/tests/logs/rule-custom-2019-03-06-1019/xccdf_org.ssgproject.content_rule_directory_access_var_log_audit.retry
```

Comment (redhatrises) [reply linux_os/guide/system/auditing/auditd_configure_rules/directory_access_var_log_audit/ansible/shared.yml]: @yuumasato huh.... no idea how that got there. Fixed.

File: linux_os/guide/system/auditing/auditd_configure_rules/directory_access_var_log_audit/ansible/shared.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+- name: Use matched file as the recipient for the rule
+  set_fact:
+    all_files:
+    - '{{ find_var_log_audit.files | map(attribute=''path'') | list | first }}'
+  when: find_var_log_audit.matched > 0
+
+- name: Inserts/replaces the /var/log/audit/ rule in rules.d
+  lineinfile:
+    path: '{{ all_files[0] }}'
+    line: -a always,exit -F dir=/var/log/audit/ -F perm=r -F auid&gt;={{{ auid }}} -F auid!=unset
```
Comment (yuumasato) [inline linux_os/guide/system/auditing/auditd_configure_rules/directory_access_var_log_audit/ansible/shared.yml]: I'm sorry I was not explicit, these `>` should not be escaped too.

File: linux_os/guide/system/auditing/auditd_configure_rules/directory_access_var_log_audit/ansible/shared.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+      -F key=access-audit-trail
+    create: true
+
+- name: Inserts/replaces the /var/log/audit/ rule in audit.rules
+  lineinfile:
+    path: /etc/audit/audit.rules
+    line: -a always,exit -F dir=/var/log/audit/ -F perm=r -F auid&gt;={{{ auid }}} -F auid!=unset
```
Comment (yuumasato) [inline linux_os/guide/system/auditing/auditd_configure_rules/directory_access_var_log_audit/ansible/shared.yml]: I'm sorry I was not explicit, these `>` should not be escaped too.

File: linux_os/guide/system/auditing/auditd_configure_rules/directory_access_var_log_audit/ansible/shared.yml (hunk 3 above)
Comment (jan-cerny) [reply linux_os/guide/system/auditing/auditd_configure_rules/directory_access_var_log_audit/ansible/shared.yml]: done

File: linux_os/guide/system/auditing/auditd_configure_rules/directory_access_var_log_audit/ansible/shared.yml (hunk 4 above)
Comment (jan-cerny) [reply linux_os/guide/system/auditing/auditd_configure_rules/directory_access_var_log_audit/ansible/shared.yml]: done

Comment (redhatrises) [reply linux_os/guide/system/auditing/auditd_configure_rules/directory_access_var_log_audit/ansible/shared.yml]: @yuumasato you probably didn't need to be explicit. I thought I had made sure that all instances of `&gt;` where fixed. I guess not. Thanks and fixed.

File: linux_os/guide/system/auditing/auditd_configure_rules/directory_access_var_log_audit/ansible/shared.yml (hunk 3 above)
Comment (redhatrises) [reply linux_os/guide/system/auditing/auditd_configure_rules/directory_access_var_log_audit/ansible/shared.yml]: @yuumasato fixed

**Comment by yuumasato** [comment]
@redhatrises great, thanks.

**Pull request merged**